### DISCUSSION TITLE
cheats not working

### DISCUSSION THREAD

**Discussion by 44brabal** [thenick775/gbajs3#107] [General]
plz help

**Comment by thenick775** [comment]
What steps have you taken, what cheats have you entered, and what rom are you attempting to enter a cheat into?

I will do some validation against games I currently am playing to validate existing behavior.

I'll need more information to assist you.

**Comment by thenick775** [comment]
I've confirmed cheats are working as in the spec:

https://github.com/thenick775/gbajs3/assets/40526638/622b634f-24a8-4bef-8ddb-45b95d85d167

Note: 
- multiline cheats must be separated with `+` as you can see here
- enable must be checked for the cheat to take effect
- all cheat types used by MGBA are supported, where raw uploaded files must be in libretro format only
- raw cheats are in libretro format, if using the individual cheat entry view, you don't have to worry about that

Context:

Cheat I entered in the video:
```
82026E14 0059+82026E16 0384
```

Full raw cheats list on my current rom:
```
cheats = 3

cheat0_desc = "Soothe bell"
cheat0_enable = false
cheat0_code = "82026E14 00f5+82026E16 0384"

cheat1_desc = "Ability pill"
cheat1_enable = false
cheat1_code = "82026E18 0044+82026E1A 03E7"

cheat2_desc = "Pinap berry"
cheat2_enable = true
cheat2_code = "82026E14 0059+82026E16 0384"

```

This cheat was sourced from my list(s) located [here](https://github.com/thenick775/miscellaneous-scripts-bin/blob/master/retroarch_utilities/amberelec_cheats_assembly/cheats/inclement_emerald.cht).

**Reply by woodj3wh8r** [reply]
Was looking for a solution first video is literally for the game I wanted to cheat in thank you.

**Comment by thenick775** [comment]
I'm going to convert this to a discussion, as I have validated cheats are fully functional as far as I can see, and additionally I'd like to assist you in getting your cheats working.

If we find an actual problem that requires a code change, I'll reopen this as an issue.

But once I have the information from the first comment, I can assist you in getting your cheats working! 😎
